### PR TITLE
[tf.keras] Add custom scalar tensorboard summary for metrics

### PR DESCRIPTION
**Pull request by lgeiger**
It can be very useful to view training and validation loss in a single chart inside tensorboard.

Currently this is only easily possible by using two different summary writers that write to different tags: https://stackoverflow.com/questions/47877475/keras-tensorboard-plot-train-and-validation-scalars-in-a-same-figure/48393723

This PR extends the keras callback to display trainings and validation metrics in a single chart using the [custom scalar plugin](https://github.com/tensorflow/tensorboard/tree/master/tensorboard/plugins/custom_scalar):

tb_custom

Note: https://github.com/tensorflow/tensorboard/pull/1294 is required to properly display multiple runs.

### CODE REVIEW

**Comment by lgeiger** [comment]
4921064dd535d84aa031f8116e583b151dd46e97 introduced batch-level metrics. The latest commit modifies this PR to work with the latest master:
localhost_6006_ 1

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. The code looks good to me. Is this covered by current unit tests? (if not, please add a unit test).

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmlarsen** [comment]
@lgeiger Could you please add a unit test?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by lgeiger** [comment]
> Could you please add a unit test?

I'll take a look at it next week.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmlarsen** [comment]
@lgeiger any updates?

**Comment by lgeiger** [comment]
I'll try to revisit this PR soon, to support tf@2.0

**Comment by lgeiger** [comment]
In TF@2.0, Keras uses two TensorBoard writers for eval and train so this PR isn't needed anymore. Closing.
